hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -205,31 +205,19 @@ private HoodieCleanMetadata runClean(HoodieTable<T, I, K, O> table, HoodieInstan
           Option.of(timer.endTimer()),
           cleanStats
       );
-      writeMetadata(metadata);
+      if (!skipLocking && config.isMetadataTableEnabled()) {
+        this.txnManager.beginTransaction(Option.empty(), Option.empty());
+      }
+      writeTableMetadata(metadata);
```

Comment:
This should not be invoked if metadata table is disabled?  It is not invoked in the original code if metadata table is disabled.

Reply:
@nsivabalan After looking at `writeTableMetadata()` method, it looks like that the method is a no-op if metadata table is disabled.  Just to confirm if that's the case.

Reply:
yes, you are right. it will be no-op.